api-compact.yaml

```diff
@@ -85,9 +91,11 @@ x-amazon-apigateway-integration: &apigateway_integration
   type: "http"
   requestParameters:
     integration.request.header.Authorization: "method.request.header.Authorization"
+    integration.request.header.Host: "${stageVariables.enterprise_host}"
```

Comment:
I think best way forward would be to add new stage variable named `gateway_host` (with api gateway's domain) and use this to forward `Host` headers.

@mattdrayer , @douglashall , @efagin -- Thoughts ?